src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -4574,7 +4574,18 @@ def _main_program_handler(self, node, parent):
         :returns: PSyIR representation of node.
         :rtype: :py:class:`psyclone.psyir.nodes.Routine`
 
+        :raises NotImplementedError: if the node contains a Contains clause.
         '''
+        try:
+            _first_type_match(node.children,
+                              Fortran2003.Internal_Subprogram_Part)
+            has_contains = True
+        except ValueError:
+            has_contains = False
+        if has_contains:
+            raise NotImplementedError("Psyclone doesn't yet support 'Contains'"
```

Comment:
Please could you do away with the `has_contains` and just move this raise directly into L4582. L4594 would then just be `pass` (please also add a comment along the lines of "The routine does not have a Contains section").

Reply:
Done, also fixed typo in the exception.